geode-connectors/src/main/java/org/apache/geode/connectors/jdbc/internal/cli/CreateMappingPreconditionCheckFunction.java

```diff
@@ -0,0 +1,245 @@
+        JDBCType jdbcType = tableMetaData.getColumnDataType(jdbcName);
+        FieldMapping fieldMapping =
+            createFieldMapping(jdbcName, jdbcType.getName(), isNullable, pdxFields);
+        fieldMappings.add(fieldMapping);
+      }
+      if (regionMapping.getIds() == null || regionMapping.getIds().isEmpty()) {
+        List<String> keyColumnNames = tableMetaData.getKeyColumnNames();
+        output[0] = String.join(",", keyColumnNames);
+      }
+      String member = context.getMemberName();
+      return new CliFunctionResult(member, output);
+    } catch (SQLException e) {
+      throw JdbcConnectorException.createException(e);
+    }
+  }
+
+  private FieldMapping createFieldMapping(String jdbcName, String jdbcType, boolean jdbcNullable,
+      List<PdxField> pdxFields) {
+    String pdxName = null;
+    String pdxType = null;
+    for (PdxField pdxField : pdxFields) {
+      if (pdxField.getFieldName().equals(jdbcName)) {
+        pdxName = pdxField.getFieldName();
+        pdxType = pdxField.getFieldType().name();
+        break;
+      }
+    }
+    if (pdxName == null) {
+      // look for one inexact match
+      for (PdxField pdxField : pdxFields) {
+        if (pdxField.getFieldName().equalsIgnoreCase(jdbcName)) {
+          if (pdxName != null) {
+            throw new JdbcConnectorException(
+                "More than one PDX field name matched the column name \"" + jdbcName + "\"");
+          }
+          pdxName = pdxField.getFieldName();
+          pdxType = pdxField.getFieldType().name();
+        }
+      }
+    }
+    if (pdxName == null) {
+      throw new JdbcConnectorException(
+          "No PDX field name matched the column name \"" + jdbcName + "\"");
+    }
+    return new FieldMapping(pdxName, pdxType, jdbcName, jdbcType, jdbcNullable);
+  }
+
+  private PdxType getPdxTypeForClass(InternalCache cache, TypeRegistry typeRegistry,
+      String className) {
+    Class<?> clazz = loadPdxClass(className);
+    PdxType result = typeRegistry.getExistingTypeForClass(clazz);
+    if (result != null) {
+      return result;
+    }
+    return generatePdxTypeForClass(cache, typeRegistry, clazz);
+  }
+
+  /**
+   * Generates and returns a PdxType for the given class.
+   * The generated PdxType is also stored in the TypeRegistry.
+   *
+   * @param cache used to generate pdx type
+   * @param clazz the class to generate a PdxType for
+   * @return the generated PdxType
+   * @throws JdbcException if a PdxType can not be generated
+   */
+  private PdxType generatePdxTypeForClass(InternalCache cache, TypeRegistry typeRegistry,
+      Class<?> clazz) {
+    Object object = createInstance(clazz);
+    if (PdxSerializable.class.isAssignableFrom(clazz)) {
+      try {
+        cache.registerPdxMetaData(object);
+      } catch (SerializationException ex) {
+        throw new JdbcConnectorException(
+            "Could not generate a PdxType for the class " + clazz.getName() + " because: " + ex);
+      }
+    } else {
```

Comment:
It now tries to register the pdx meta data. If that fails (because pdx serialization is not configured for that class) then it attempts to serialize using its own ReflectionBasedAutoSerializer.